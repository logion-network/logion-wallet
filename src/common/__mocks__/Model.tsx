import {
    fetchProtectionRequests,
    preVoid,
    isGrantedAccess,
} from './ModelMock';

export {
    fetchProtectionRequests,
    preVoid,
    isGrantedAccess,
};
