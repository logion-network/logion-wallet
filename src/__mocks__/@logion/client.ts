import { CheckResultType } from "@logion/client/dist/CollectionItem.js";
import { DefaultSignAndSendStrategy } from "@logion/client/dist/Signer.js";

import {
    LogionClient,
    AccountTokens,
    toIsoString,
    fromIsoString,
    ClosedCollectionLoc,
    DraftRequest,
    EditableRequest,
    OpenLoc,
    LocRequestState,
    LocsState,
    HashOrContent,
} from '../LogionClientMock';

export {
    LogionClient,
    AccountTokens,
    toIsoString,
    fromIsoString,
    CheckResultType,
    ClosedCollectionLoc,
    DraftRequest,
    EditableRequest,
    OpenLoc,
    DefaultSignAndSendStrategy,
    LocRequestState,
    LocsState,
    HashOrContent,
}

export { MimeType } from "@logion/client/dist/Mime.js";
