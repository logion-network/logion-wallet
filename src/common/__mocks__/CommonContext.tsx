import { 
    useCommonContext as useCommonContextMock,
} from './CommonContextMock';

export const useCommonContext = useCommonContextMock;
export { ExpectNewTransactionStatus } from "./CommonContextMock";
