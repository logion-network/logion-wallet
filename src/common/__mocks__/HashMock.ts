export let sha256Hex: jest.Mock<any, any> = jest.fn();

export function setExpectedHash(expectedHash: string) {
    sha256Hex = jest.fn().mockResolvedValue(expectedHash)
}
