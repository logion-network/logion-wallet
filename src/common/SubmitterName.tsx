import { LocData, VerifiedThirdParty } from "@logion/client";
import { Cell } from "./Table";
import LegalOfficerName from "./LegalOfficerNameCell";
import Icon from "./Icon";
import OverlayTrigger from "react-bootstrap/OverlayTrigger";
import Tooltip from "react-bootstrap/Tooltip";

export interface Props {
    loc: LocData
    submitter: string | undefined;
}

interface Identity {
    firstName: string;
    lastName: string;
}

function format(identity: Identity | undefined, address: string | undefined): string {
    return identity ? identity.firstName + " " + identity.lastName : address || "-";
}

function findVTP(loc: LocData, address: string): VerifiedThirdParty | undefined {
    return loc.issuers.find(vtp => vtp.address === address);
}

export default function SubmitterName(props: Props) {
    const { loc, submitter } = props;
    if (submitter === loc.ownerAddress) {
        return (
            <LegalOfficerName address={ submitter } />
        )
    } else {
        const identity = getIdentity(submitter, loc);
        return (
            <Cell content={
                <div>
                    { format(identity, submitter) }
                    {
                        submitter && submitter !== loc.requesterAddress &&
                        <VTPBadge />
                    }
                </div>
            }
            />
        )
    }
}

function getIdentity(submitter: string | undefined, loc: LocData): Identity | undefined {
    if(submitter === loc.requesterAddress) {
        return loc.userIdentity;
    } else if(submitter) {
        return findVTP(loc, submitter);
    } else {
        return undefined;
    }
}

function VTPBadge() {
    return (
        <OverlayTrigger
            placement="bottom-end"
            delay={ 500 }
            overlay={
                <Tooltip id={ Math.random().toString() }>
                    contributed by this Verified Issuer at the request of the Legal Officer in charge
                </Tooltip>
            }
        >
            <span>
                &nbsp;
                <Icon icon={ { id: "vtp-badge" } } />
            </span>
        </OverlayTrigger>
    )
}
