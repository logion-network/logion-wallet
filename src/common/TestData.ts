import { LegalOfficer, PostalAddress, UserIdentity } from '@logion/client';
import { Transaction } from '@logion/client/dist/TransactionClient.js';
import { Coin, CoinBalance, SYMBOL } from '@logion/node-api/dist/Balances.js';
import { PrefixedNumber, ATTO } from '@logion/node-api/dist/numbers.js';

import { ColorTheme, rgbaToHex } from './ColorTheme';

export const DEFAULT_LEGAL_OFFICER = "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY"; // Alice
export const ANOTHER_LEGAL_OFFICER = "5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty"; // Bob
export const A_THIRD_LEGAL_OFFICER = "5FLSigC9HGRKVhB9FiEo4Y3koPsNmBmLJbpXg2mp1hXcS59Y"; // Charlie

export const DEFAULT_IDENTITY: UserIdentity = {
    firstName: "John",
    lastName: "Doe",
    email: "john.doe@logion.network",
    phoneNumber: "+1234",
};

export const DEFAULT_ADDRESS: PostalAddress = {
    line1: "Place de le République Française, 10",
    line2: "boite 15",
    postalCode: "4000",
    city: "Liège",
    country: "Belgium",
};

export const COLOR_THEME: ColorTheme = {
    type: 'light',
    shadowColor: rgbaToHex('#3b6cf4', 0.1),
    dashboard: {
        background: '#eff3fe',
        foreground: '#000000',
    },
    sidebar: {
        background: '#ffffff',
        foreground: '#000000',
        activeItemBackground: '#d8e2fd',
    },
    accounts: {
        iconBackground: '#3b6cf4',
        hintColor: rgbaToHex('#000000', 0.6),
        foreground: '#000000',
        background: 'eff3fe',
        legalOfficerIcon: {
            category: 'legal-officer',
            id: 'account-shield'
        }
    },
    frame: {
        background: '#ffffff',
        foreground: '#000000',
        link: rgbaToHex('#3b6cf4', 0.15),
        altBackground: '#eff3fe',
    },
    topMenuItems: {
        iconGradient: {
            from: '#3b6cf4',
            to: '#6050dc',
        }
    },
    bottomMenuItems: {
        iconGradient: {
            from: '#7a90cb',
            to: '#3b6cf4',
        }
    },
    recoveryItems: {
        iconGradient: {
            from: '#7a90cb',
            to: '#3b6cf4',
        }
    },
    buttons: {
        secondary: {
            background: '#eff3fe',
            foreground: '#000000',
        }
    },
    select: {
        background: '#ffffff',
        foreground: '#000000',
        menuBackgroundColor: '#ffffff',
        selectedOptionBackgroundColor: '#ffffff',
    },
    table: {
        background: '#0c163d',
        foreground: '#ffffff',
        header: {
            background: '#0c163d',
            foreground: '#ffffff',
        },
        row: {
            background: '#152665',
            foreground: '#ffffff',
        }
    },
    tabs: {
        background: '#0c163d',
        foreground: '#ffffff',
        inactiveTab: {
            background: '#d8e2fd',
            foreground: '#ffffff',
        },
        borderColor: '#3b6cf4',
    },
    dialog: {
        background: '#eff3fe',
        foreground: '#000000',
        borderColor: '#e6007a',
    },
    dialogTable: {
        background: '#0c163d',
        foreground: '#ffffff',
        header: {
            background: '#0c163d',
            foreground: '#ffffff',
        },
        row: {
            background: '#152665',
            foreground: '#ffffff',
        }
    },
};

export const DEFAULT_COIN: Coin = {
    id: 'lgnt',
    iconId: 'lgnt',
    iconType: 'svg',
    name: "Logion",
    symbol: SYMBOL
};

export const DEFAULT_COIN_BALANCE: CoinBalance = {
    coin: DEFAULT_COIN,
    balance: new PrefixedNumber("42", ATTO),
    available: new PrefixedNumber("42", ATTO),
    level: 0.1,
};

export const DEFAULT_TRANSACTION: Transaction = {
    id: "id",
    from: "from",
    to: "to",
    pallet: "pallet",
    method: "method",
    transferValue: "420",
    tip: "0",
    fee: "125000",
    reserved: "0",
    total: "125420",
    createdOn: "2021-07-28T12:30:00.000",
    type: "Sent",
    transferDirection: "Sent",
    successful: true
}

export const DEFAULT_FAILED_TRANSACTION:Transaction = {
    ...DEFAULT_TRANSACTION,
    successful: false,
    error: {
        section: 'aSection',
        name: 'aName',
        details: 'someDetails'
    }
}

export const PATRICK: LegalOfficer = {
    name: "Patrick Gielen",
    address: DEFAULT_LEGAL_OFFICER,
    additionalDetails: "",
    userIdentity: {
        firstName: "Patrick",
        lastName: "Gielen",
        email: "patrick@logion.network",
        phoneNumber: "+32 498 237 107",
    },
    postalAddress: {
        company: "MODERO",
        line1: "Huissier de Justice Etterbeek",
        line2: "Rue Beckers 17",
        postalCode: "1040",
        city: "Etterbeek",
        country: "Belgique"
    },
    node: "http://logion.patrick.com",
    logoUrl: "",
    nodeId: "",
};

export const GUILLAUME: LegalOfficer = {
    name: "Guillaume Grain",
    address: ANOTHER_LEGAL_OFFICER,
    additionalDetails: "",
    userIdentity: {
        firstName: "Patrick",
        lastName: "Gielen",
        email: "g.grain@adrastee-lyon.fr",
        phoneNumber: "+33 4 78 52 87 56",
    },
    postalAddress: {
        company: "SELARL ADRASTEE",
        line1: "Gare des Brotteaux",
        line2: "14, place Jules Ferry",
        postalCode: "69006",
        city: "LYON",
        country: "France"
    },
    node: "http://logion.guillaume.com",
    logoUrl: "",
    nodeId: "",
};

export const ALAIN: LegalOfficer = {
    name: "Alain Barland",
    address: A_THIRD_LEGAL_OFFICER,
    additionalDetails: "",
    userIdentity: {
        firstName: "Patrick",
        lastName: "Gielen",
        email: "alain.barland@auxilia-conseils.com",
        phoneNumber: "+33 2 48 67 50 50",
    },
    postalAddress: {
        company: "AUXILIA CONSEILS 18",
        line1: "Huissiers de Justice associés",
        line2: "7 rue Jean Francois Champollion Parc Comitec",
        postalCode: "18000",
        city: "Bourges",
        country: "France"
    },
    node: "http://logion.alain.com",
    logoUrl: "",
    nodeId: "",
};

export const legalOfficers: LegalOfficer[] = [
    PATRICK,
    GUILLAUME,
    ALAIN
];
