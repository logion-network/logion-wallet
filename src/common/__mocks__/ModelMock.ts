import { LocRequest } from "@logion/client";
import { CreateLocRequest } from "../Model";

export let fetchProtectionRequests = jest.fn();

export function setFetchProtectionRequests(mockFn: any) {
    fetchProtectionRequests = mockFn;
}

export async function createLocRequest(axios: any, request: CreateLocRequest): Promise<LocRequest> {
    return Promise.resolve({
        ...request,
        createdOn: "2022-01-20T09:24:00.000",
        description: "Test",
        files: [],
        id: "671e65cf-42c3-435b-a222-be008e7fb3d1",
        links: [],
        metadata: [],
        status: "OPEN",
        verifiedThirdParty: false,
        selectedIssuers: [],
    });
}

export const preVoid = jest.fn();

export function resetDefaultMocks() {
    preVoid.mockResolvedValue(undefined);
}

export function isGrantedAccess() {
    return true;
}
