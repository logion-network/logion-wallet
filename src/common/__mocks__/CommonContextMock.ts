import { COLOR_THEME, threeLegalOfficers } from '../TestData';
import { CommonContext, Viewer } from "../CommonContext";
import { BalanceState } from "@logion/client/dist/Balance.js";

export let setColorTheme = jest.fn();

export let refresh = jest.fn();

export let balanceState: BalanceState | undefined = {
    balances: [],
} as unknown as BalanceState;

export function setBalanceState(value: BalanceState | undefined) {
    balanceState = value;
}

let viewer: Viewer = "User";

export function setViewer(_viewer: Viewer) {
    viewer = _viewer;
}

export enum ExpectNewTransactionStatus {
    IDLE,
    WAITING_NEW_TRANSACTION,
    DONE
}

export function useCommonContext() {
    const commonContext:Partial<CommonContext> = {
        balanceState,
        accountsBalances: {},
        colorTheme: COLOR_THEME,
        setColorTheme,
        refresh,
        nodesUp: [],
        nodesDown: [],
        availableLegalOfficers: threeLegalOfficers,
        viewer,
        backendConfig: () => ({ features:{ iDenfy: false, vote: false }}),
        expectNewTransactionState: { status: ExpectNewTransactionStatus.IDLE, refreshCount: 0 },
    };
    return commonContext;
}

