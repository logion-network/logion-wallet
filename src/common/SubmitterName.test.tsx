import { shallowRender } from "../tests";
import SubmitterName from "./SubmitterName";
import { DEFAULT_LEGAL_OFFICER, DEFAULT_IDENTITY } from "./TestData";
import { mockValidPolkadotAccountId } from 'src/__mocks__/LogionMock';
import { LocData } from "@logion/client";

jest.mock('../logion-chain');

describe("SubmitterName", () => {

    const REQUESTER = mockValidPolkadotAccountId("5H17oaxYsRpV18tYjMMbozN84JAVgG6MH5pNWjisGYDtc9WR");
    const VTP = "5DtGNqi7yp5TqRcPckYVZh5XqdaSKE9rsDur5xur4tkmBAhB";
    const loc = {
        ownerAddress: DEFAULT_LEGAL_OFFICER.address,
        requesterAddress: REQUESTER,
        userIdentity: DEFAULT_IDENTITY,
        issuers: [ {
            address: VTP,
            firstName: "Scott",
            lastName: "Tiger",
        } ]
    } as unknown as LocData;

    it("renders when submitter is Legal Officer", () => {
        const tree = shallowRender(<SubmitterName loc={ loc } submitter={ DEFAULT_LEGAL_OFFICER.address } />);
        expect(tree).toMatchSnapshot()
    })

    it("renders when submitter is Requester", () => {
        const tree = shallowRender(<SubmitterName loc={ loc } submitter={ REQUESTER.address } />);
        expect(tree).toMatchSnapshot()
    })

    it("renders when submitter is VTP", () => {
        const tree = shallowRender(<SubmitterName loc={ loc } submitter={ VTP } />);
        expect(tree).toMatchSnapshot()
    })
})
