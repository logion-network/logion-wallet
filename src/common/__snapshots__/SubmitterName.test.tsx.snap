// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitterName renders when submitter is Legal Officer 1`] = `
<LegalOfficerName
  address="5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY"
/>
`;

exports[`SubmitterName renders when submitter is Requester 1`] = `
<Cell
  content={
    <div>
      John Doe
    </div>
  }
/>
`;

exports[`SubmitterName renders when submitter is VTP 1`] = `
<Cell
  content={
    <div>
      Scott Tiger
      <VTPBadge />
    </div>
  }
/>
`;
